ddl/ddl_api.go

```diff
@@ -1120,6 +1127,10 @@ func (d *ddl) AddColumn(ctx sessionctx.Context, ti ast.Ident, spec *ast.AlterTab
 		return errors.Trace(infoschema.ErrTableNotExists.GenByArgs(ti.Schema, ti.Name))
 	}
 
+	if err = checkAddColumnTooManyColumns(len(t.Meta().Columns), spec.NewColumns); err != nil {
```

Comment:
we need to check the columns count in onAddColumn stage. Imagine parallel adding column, can lead to TiDB exceed the limit.